index.js

```diff
@@ -78,17 +79,22 @@ export class StuckTransactionsCanceller {
 
   async removeConfirmed (tx) {
     assert.strictEqual(typeof tx.hash, 'string')
-    await this.#store.remove(tx.hash)
+    const txs = await this.#store.list()
+    for (const _tx of txs) {
+      if (_tx.nonce === tx.nonce) {
+        await this.#store.remove(_tx.hash)
+      }
+    }
   }
 
   async cancelOlderThan (ageMs) {
     assert.strictEqual(typeof ageMs, 'number')
 
     this.#log('Checking for stuck transactions...')
     const txs = await this.#store.list()
-    const txsToCancel = txs.filter(tx => {
-      return new Date() - new Date(tx.timestamp) > ageMs
-    })
+    const txsToCancel = txs
+      .filter(tx => new Date() - new Date(tx.timestamp) > ageMs)
+      .filter(tx => !txs.find(_tx => _tx.nonce === tx.nonce && _tx.gasLimit > tx.gasLimit))
```

Comment:
Would you mind adding a code comment to explain the intent of this condition?

```suggestion
      // Ignore transactions that were already replaced
      .filter(tx => !txs.find(_tx => _tx.nonce === tx.nonce && _tx.gasLimit > tx.gasLimit))
```